monkey/monkey_island/cc/environment/utils.py

```diff
@@ -35,10 +37,14 @@ def _create_secure_directory(path: str, create_parent_dirs: bool):
         raise ex
 
 
-def set_secure_permissions(dir_path: str):
+def _create_secure_directory_windows(path: str):
     try:
-        if is_windows_os():
-            windows_permissions.set_perms_to_owner_only(folder_path=dir_path)
+        security_attributes = win32security.SECURITY_ATTRIBUTES()
+        security_attributes.SECURITY_DESCRIPTOR = windows_permissions.get_sd_for_owner_only_perms()
+        win32file.CreateDirectory(path, security_attributes)
     except Exception as ex:
-        LOG.error(f"Permissions could not be set successfully for {dir_path}: {str(ex)}")
+        LOG.error(
+            f'Could not create a directory at "{path}" (maybe environmental variables could not be '
+            f"resolved?): {str(ex)}"
```

Comment:
There are lots of reasons directory creation might fail. It should be obvious that environment variables weren't resolved when the "{path}" is printed out as part of this error message. Adding this suggestion might send the user on a wild goose chase.

```suggestion
            f'Could not create a directory at "{path}": {str(ex)}"
```